services/npm/npm-rating.service.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+
+const Joi = require('joi')
+const { starRating } = require('../text-formatters')
+const { BaseJsonService } = require('..')
+
+// https://api-docs.npms.io/#api-Package-GetPackageInfo
+const responseSchema = Joi.object({
+  score: Joi.object({
+    final: Joi.number().required(),
+    detail: Joi.object({
+      maintenance: Joi.number().required(),
+      popularity: Joi.number().required(),
+      quality: Joi.number().required(),
+    }),
+  }),
+}).required()
+
+module.exports = class NpmRating extends BaseJsonService {
+  static category = 'rating'
```

Comment:
I don't feel like this is the right category. Rating badges are really intended to reflect the aggregate picture of reviews/ratings provided directly users, whereas npms is really more of a quantitative analysis that takes into account various attributes of a package. This are much more akin to quality and accordingly `analysis`feels a better fit